### PR TITLE
feat: adds react-query automatic proxy package

### PR DESCRIPTION
**Pull request by stalniy**
## Why

to have a way to automatically convert service into react-query observable hooks

## Summary by CodeRabbit

* **New Features**
  * Added a new react-query-proxy package providing a proxy-based integration to useQuery/useMutation for async SDKs, with nested support and customizable query-key mapping.

* **Documentation**
  * Added a README with installation, usage examples, query key/invalidation guidance, and license.

* **Tests**
  * Added comprehensive tests covering key generation, query/mutation flows, nested proxies, caching, and edge cases.

* **Chores**
  * Added package metadata and project configs (linting, testing, TypeScript) and simplified ESLint in the query-sdk package.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/react-query-proxy/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  alerts: {
+    async list(input?: { page?: number }) {
+      return fetchAlerts(input)
+    },
+
+    async create(input: { name: string }) {
+      return createAlert(input)
+    },
+  },
+}
```
Comment (baktun14) [inline packages/react-query-proxy/README.md:51]: question: how would this be defined? with the auto generation code from this pr? https://github.com/akash-network/console/pull/2437/changes

Comment (stalniy) [reply packages/react-query-proxy/README.md:51]: these are 2 conflicting approaches. Here is the key difference:

react-query hooks inference (this PR):
- it doesn't care how you generate SDK, it can be protobuf based sdk, REST based sdk or even self-written service. It just wraps that service and expose react-query hooks for its methods.
- we have clear layer separation: SDK on service layer and this proxy acts as a glue to expose any SDK/service to react world (i.e., in View).

qraft PR:
- works only with OpenAPI defined APIs. Cannot be applied to chain-sdk or self-written services
- no clear separation between view and service layer, since it was written to support react and react SSR. This potentially may cause react & react-query libs to leak into server side bundle (in console-api/etc). IMO, this is code "smell"

Comment (stalniy) [reply packages/react-query-proxy/README.md:51]: This is part of built console-api, you can see that it contains react-query code and very likely it also contains react. So, our API depends on react without a good reason to do so. Who knows which surprises this can bring in long term

Screenshot 2026-01-06 at 08 34 58

**Review by ygrishajev — APPROVED** [review]
